client/keys/show.go

```diff
@@ -76,21 +104,35 @@ func getKey(name string) (keys.Info, error) {
 func GetKeyRequestHandler(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	name := vars["name"]
+	bechPrefix := r.URL.Query().Get(FlagBechPrefix)
+
+	if bechPrefix == "" {
+		bechPrefix = "acc"
+	}
```

Comment:
We could return a 400 or 422, but that leads to the following points:
* It doesn't keep inline with the analogous CLI command (where `acc` is the default)
* It breaks backwards compatibility -- voyager team and users now have to always specify `bech?=...`.

I don't see why having `acc` as the default is bad here.

Reply:
> It doesn't keep inline with the analogous CLI command (where acc is the default)

But the CLI will fail if you submit "" as the bech flag input - so it's actually the reverse of what you're saying - by allowing an empty string to be submitted we are deviating the REST client from the CLI. I guess the fundamental difference is that the CLI by default includes flag defaults (which can be viewed with the help command) where as we can't do that besides hard coding them in for the REST API - we don't really have a well documented area to look for rest defaults besides looking at the code directly

> It breaks backwards compatibility -- voyager team and users now have to always specify bech?=...

I think we should when appropriate break our APIs, and just be explicit about it in the changelogs - we shouldn't not break something just for the sake of not breaking alone 

> I don't see why having acc as the default is bad here.

I guess this is the real question - I'm guessing that although creating a default may be expedient it may lead to misunderstanding on what the command is doing, aka whether you're retrieving the `acc` or the `val` is a parallel step, not subsequent. Not totally sure

Reply:
Valid points -- it really boils down to UX I think. Also, I'm completely not against breaking things, but I just also don't want to do it willy nilly without thinking of other users, hence my initial hesitation.

I'm happy to make the query param required. Would like thoughts from @faboweb.

Reply:
yeah let's hear what @faboweb thinks

Reply:
Sorry guys, skipped this convo.

> I'm completely not against breaking things, but I just also don't want to do it willy nilly without thinking of other users

Yes, yes, yes! :)

---

The alignment is, that the REST client is not targeted at validators. We want validators to only use the CLI. A REST user will therefor not want a validator or consensus address from the keystore so we can for the LCD safely assume querying for an `acc` for all I know.

Reply:
OK -- leaving the logic as is then. Thanks @faboweb